dbt_platform_helper/providers/cloudformation.py

```
@@ -8,7 +8,7 @@
 
 
 class CloudFormation:
-    def __init__(self, cloudformation_client, iam_client, ssm_client):
+    def __init__(self, cloudformation_client, iam_client=None, ssm_client=None):
```

Not sure I like being able to initialise this class with or without iam and ssm clients, though I see these clients are only needed for certain functionalities... maybe a better pattern here would be to pass in the session and then the clients can be created when needed.  Something to return to when we consolidate the way we're handling boto3 stuff I guess.  Maybe could add a TODO?  Or at least we should raise an exception if one of the functions is called without the client being there?  e.g. if we try CloudFormation(cloudformation_client).add_stack_delete_policy_to_task_role